app/Http/Traits/DataAccessApplicationHelpers.php

```diff
@@ -418,4 +418,31 @@ public function groupApplicationsByProject(LengthAwarePaginator $applications):
             $page
         );
     }
+
+    public function returnApplicationsInProject(LengthAwarePaginator $applications): LengthAwarePaginator
+    {
+        $applicationsResult = array();
+
+        foreach ($applications as $application) {
+            $applicationArray = array();
+            $sameProject = DataAccessApplication::where('project_id', $application['project_id'])
+                ->get();
+            $teams = array();
+            foreach ($sameProject as $app) {
+                $teams = array_merge($teams, $app['teams']->toArray());
+            }
+            $application['teams'] = $teams;
+            $applicationsResult[] = $application;
+        }
+
+        $page = $applications::resolveCurrentPage();
+        $perPage = Config::get('constants.per_page');
+
+        return new LengthAwarePaginator(
+            $applicationsResult,
+            count($applicationsResult),
+            $perPage,
+            $page
+        );
```

Comment:
Looks like `applicationsResult` will be always the same length as `applications` so dont need to reconstruct the LengthAwarePaginatior?

So you could transform instead of reconstruct:
```
    $applications->getCollection()->transform(function ($application) {
        ...
       <modify teams>
       ..
       }
      return $applications
```


Making SQL calls within the forloop isnt very optimal either, maybe pull that outside the loop:
```
$projectIds = $applications->getCollection()->pluck('project_id')->unique();
```
then can load them all via `->whereIn('project_id',$projectIds)` to lookup. I get that you might not have many applications per project though... so might be overkill.

Reply:
Good shout on using `transform`, that's much nicer!  

I don't think it is worth pulling the lookup out of the loop in this case. We're not expecting many applications per project.